SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -76,79 +141,62 @@ void ASpatialTestRepNotify::PrepareTest()
 		FinishStep();
 	});
 
 	// All clients check that the replicated variables were received correctly and that RepNotify acted as expected.
 	AddStep(
-		TEXT("SpatialTestRepNotifyAllClientsCheckValuesAndRepNotifies"), FWorkerDefinition::AllClients,
-		[this]() -> bool {
-			// First make sure that we have correctly received the replicated variables, before checking the RepNotify behaviour.
-			if (!(OnChangedRepNotifyInt1 == 10 && AlwaysRepNotifyInt1 == 20 && OnChangedRepNotifyInt2 == 30 && AlwaysRepNotifyInt2 == 40))
-			{
-				return false;
-			}
+		TEXT("SpatialTestRepNotifyAllClientsCheckValuesAndRepNotifies"), FWorkerDefinition::AllClients, nullptr, nullptr,
+		[this](float DeltaTime) {
+			// First make sure that we have correctly received the replicated variables.
+			RequireEqual_Int(TestActor->OnChangedRepNotifyInt1, 10, TEXT("Test actor's OnChangedRepNotifyInt1 property should be 10"));
+			RequireEqual_Int(TestActor->AlwaysRepNotifyInt1, 20, TEXT("Test actor's AlwaysRepNotifyInt1 property should be 20"));
+			RequireEqual_Int(TestActor->OnChangedRepNotifyInt2, 30, TEXT("Test actor's OnChangedRepNotifyInt2 property should be 30"));
+			RequireEqual_Int(TestActor->AlwaysRepNotifyInt2, 40, TEXT("Test actor's AlwaysRepNotifyInt2 property should be 40"));
 
-			if (!(TestArray.Num() == 3 && TestArray[0] == 1 && TestArray[1] == 2 && TestArray[2] == 30))
+			if (RequireEqual_Int(TestActor->TestArray.Num(), 3, TEXT("Test actor's TestArray should have 3 elements")))
 			{
-				return false;
+				RequireEqual_Int(TestActor->TestArray[0], 1, TEXT("Test actor's TestArray should have 1 at index 0"));
+				RequireEqual_Int(TestActor->TestArray[1], 2, TEXT("Test actor's TestArray should have 2 at index 1"));
+				RequireEqual_Int(TestActor->TestArray[2], 30, TEXT("Test actor's TestArray should have 30 at index 2"));
 			}
 
-			return true;
-		},
-		[this]() {
 			// At this point, we have correctly received the values of all replicated variables, therefore the RepNotify behaviour can be
 			// checked.
 
 			// Since the RepNotify for this variable is using the default REPNOTIFY_OnChanged, we expect it not to get called on the
 			// clients.
-			if (bOnRepOnChangedRepNotifyInt1Called)
-			{
-				FinishTest(EFunctionalTestResult::Failed, TEXT("OnRepOnChangedRepNotifyInt1 should not be called on the clients"));
-				return;
-			}
-
-			// In this case, the RepNotify uses REPNOTIFY_Always so we expect it to be called on the clients.
-			if (!bOnRepAlwaysRepNotifyInt1Called)
-			{
-				FinishTest(EFunctionalTestResult::Failed, TEXT("OnRepOnChangedRepNotifyInt1 should be called on the clients"));
-				return;
-			}
+			RequireEqual_Bool(TestActor->bOnRepOnChangedRepNotifyInt1Called, false,
+							  TEXT("Test actor's OnChangedInt1 Rep Notify should not be called as the value stayed the same"));
+			RequireEqual_Bool(TestActor->bOnRepAlwaysRepNotifyInt1Called, true,
+							  TEXT("Test actor's AlwaysInt1 Rep Notify should be called"));
```

Comment:
I guess this is my fault, but this could be a bit more verbose, saying it should have been called even though the value stayed the same because this is an always-repnotify var.

Reply:
yee, updated